internal/dataplane/parser/translators/ingress_atc.go

```diff
@@ -238,3 +239,178 @@ func sniMatcherFromSNIs(snis []string) atc.Matcher {
 	}
 	return atc.Or(matchers...)
 }
+
+type IngressRoutePriorityTraits struct {
+	MatchFields   int
+	PlainHostOnly bool
+	HeaderCount   int
+	MaxPathLength int
+	HasRegexPath  bool
+}
+
+// EncodeToPriority encodes the traits to `priority` field used in Kong expression based routes.
+func (t IngressRoutePriorityTraits) EncodeToPriority() int {
```

Comment:
I think it could be useful to have a kind of an ASCII-art style explanation of the encoding, something like:

```go
//        4                   3                   2                   1
//  3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
// | MF  | Header Number |P|         PRESERVED           |R|          Path Length          |
// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
//
// Where:
//  - MF (Match Fields): how many fields there are to match on.
//  - Header Number: number of headers to match.
//  - PRESERVED: reserved for future use? Why?
//  - P (Plain Host): if set, ALL hosts are non-wildcard.
//  - R (Regex): if set, regex match is used.
//  - Path Length: maximum length of the path to match.
```

WDYT?

Reply:
Done.